azext_iot/operations/hub.py

```diff
@@ -1302,9 +1313,13 @@ def _iot_hub_configuration_create(
                 "metrics json must include the '{}' property".format(metrics_key)
             )
         metrics = metrics[metrics_key]
+    elif custom_metric_queries:
+        metrics = validate_key_value_pairs(";".join(custom_metric_queries))
```

Comment:
There is a function intended for nargs kvp processing, its called `assemble_nargs_to_dict`, that is currently in the deviceupdate namespace area, that can be extracted and put somewhere more centric like our common utility module.

That being said, the above is being done in this PR https://github.com/Azure/azure-iot-cli-extension/pull/591